gcloud-java-datastore/src/test/java/com/google/gcloud/datastore/DatastoreTest.java

```
@@ -404,6 +406,42 @@ public void testRunGqlQueryWithCasting() {
   }
 
   @Test
+  public void testGqlQueryPagination() {
+    DatastoreRpcFactory rpcFactoryMock = EasyMock.createStrictMock(DatastoreRpcFactory.class);
+    DatastoreRpc rpcMock = EasyMock.createStrictMock(DatastoreRpc.class);
+    EasyMock.expect(rpcFactoryMock.create(EasyMock.anyObject(DatastoreOptions.class)))
+        .andReturn(rpcMock);
+    List<com.google.datastore.v1beta3.RunQueryResponse> responses =
+        buildResponsesForQueryPagination();
+    for (int i = 0; i < responses.size(); i++) {
+      try {
+        EasyMock
+            .expect(rpcMock.runQuery(
+                EasyMock.anyObject(com.google.datastore.v1beta3.RunQueryRequest.class)))
+            .andReturn(responses.get(i));
+      } catch (DatastoreRpcException e) {
+        fail("Unexpected DatastoreRpcException");
```

Is that better than just let the exception propagate from the method?